shell/common/engine.cc

```diff
@@ -234,11 +234,25 @@ void Engine::RunBundleAndSource(const std::string& bundle_path,
   std::string packages_path = packages;
   if (packages_path.empty())
     packages_path = FindPackagesPath(main);
-  if (!bundle_path.empty())
-    ConfigureAssetBundle(bundle_path);
-  ConfigureRuntime(GetScriptUriFromPath(main));
-  load_script_error_ =
-      runtime_->dart_controller()->RunFromSource(main, packages_path);
+
+  std::vector<uint8_t> platform_kernel;
+  if (!bundle_path.empty()) {
+    FXL_LOG(WARNING) << "RunBundleAndSource bundle_path is not empty";
+    GetAssetAsBuffer(blink::kPlatformKernelAssetKey, &platform_kernel);
+  }
+  ConfigureRuntime(GetScriptUriFromPath(bundle_path), platform_kernel);
```

Comment:
Maybe at some point we should see if it is possible to move the code which sets up platform_kernel to InitRuntime in runtime_init.cc

Reply:
I think we should be able to do that once we link platform kernel into the flutter engine binary. Until then, we get it from application bundle, which is too late for InitRuntime.

Reply:
So does this mean for every hot restart we have to include platform_kernel in the bundle so we take the correct path?

Reply:
We don't resend bundles on restart requests. Loaded bundle with its assets(including platform.dill) stays the same through the restarts. We only upload new app kernel file and request VM to reload/restart from that new kernel file.

Reply:
Thanks for the clarification.